core/src/Kokkos_TaskScheduler.hpp

```diff
@@ -44,6 +50,9 @@ static_assert(false,
 //----------------------------------------------------------------------------
 //----------------------------------------------------------------------------
 
+// We allow using deprecated classes in this file
+KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()
```

Comment:
Did you do some experiments on the compiler explorer to check that this does not inhibits deprecated warnings in user code?
I will say I am not 100% sure whether that works but I also have no evidence that it does not.

Reply:
https://godbolt.org/z/n3Tc8d6Md for `g++`. `clang++` behaves better anyway.

 I also checked that we still get warnings in the test if the test files don't have `KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()` (with clang++).

Reply:
I was more worried that there are no warnings when there should be

Reply:
Ok saw your compiler explorer link